Checkout Page

metehansalih06 opened the pull request.

metehansalih06 committed:
This is the checkout page

It will show the data from the cart and the wallet as well as the sum of the items. It is not finished yet though.

metehansalih06 committed:
Merge branch 'main' into metehan-checkout-page

metehansalih06 committed:
Merge branch 'main' into metehan-checkout-page

metehansalih06 committed:
Checkout page modal

I added a modal which pops up after clicking on pay. The site needs some more changes

metehansalih06 committed:
I am working on an error

The wallet after checkout is not displayed correctly.

metehansalih06 committed:
changed the pay-button

Works like before, as I cant find the solution for the errors of fetching data from the wallet.

metehansalih06 committed:
checkout page almost finished

I cant check if the layout of the restaurant cards are displayed correctly in the box on the left hand side.

metehansalih06 committed:
Merge branch 'main' into metehan-checkout-page

ookii-tsuki reviewed (commented):
![image](https://github.com/user-attachments/assets/16788fc8-a971-4bd7-a872-7e1e09b30c7c)
NaN EUR?

Also when the server returns 402 which means insufficient funds, it will just tell you "error try again later".
Please test your code.

metehansalih06 committed:
Pay button

After clicking pay, there should either be the message that the cart is empty or the sum of all items

metehansalih06 committed:
checkout page works

The popup now shows the sum of all items. If there is no item in the cart, it will display it accordingly. After purchase, the user will be directed to the search page. Btw. as soon as I finish the cart page, I will work on minor bugs on the page like the error message.

ookii-tsuki commented:
> Also when the server returns 402 which means insufficient funds, it will just tell you "error try again later".
Please test your code.

You didnt implement that. when the user doesnt have enough funds, they get an error instead of a useful message.
the error i get is:
> Fehler beim Abschließen der Bezahlung. Bitte versuchen Sie es später erneut.

metehansalih06 commented:
I know I didnt implement it yet, as I said I will be doing it after finishing the major issues. Everything else works, doesnt it

ookii-tsuki commented:
yes I think

metehansalih06 committed:
Merge branch 'main' into metehan-checkout-page

metehansalih06 committed:
checkout page fully working

I fixed the issue with the wrong error message. The user is now directed to the search page when not authorized. The quantity and note of an item can be changed in this page now too.

metehansalih06 committed:
Payment issue fixed

It will now return the correct error message when the user doesnt have enough in his wallet.
